docs/kb/kb1001-202401upgrade.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: default
+title: 'Upgrade: 2024.01'
+permalink: /kb/upg-202401
+nav_exclude: true
+tags: ["upgrade"]
+is_kb: true
+---
+
+## On this page
+{: .no_toc .text-delta }
+1. TOC
+{:toc}
+
+# Introduction
+This document covers the upgrade process from 2023.07 up to 2024.01. The upgrade process for this specific release is not a straight forward helm upgrade and does require quite a bit of work to get it into a running state.
+
+# Technical Documentation
+## Prerequisites
+- Access to the helm chart version `2.0.0`. You may need to `helm repo update` to grab the latest.
+- Download the prerequisite zip artifact <a href="../../assets/other/2024.01-upgrade.zip" download>here</a>
+- `kubectl` and `helm`
+
+## Guide
+Included in the zipped artifact are a number of `.yaml` files along with a PowerShell upgrade script. The script is very generic and may not be usable in your environment if your instance of CluedIn is heavily modified. As a result, it should only be used as a reference point rather than something you run. The below is the manual steps.
+
+**NOTE:** Before any upgrade takes place, ensure you have taken a snapshot or **backup of all your running PVCs** in your AKS cluster. This is very important as databases will be upgraded during this process.
+
+### Preparation
+1. The first step is to export your current running helm values files. Keep one for this process, and also keep one as a backup.
+
+**Note**: You can export by running `helm get values cluedin-platform -n cluedin --output yaml > /path/to/file.yaml`. This will export the currently connected kubeconfig.
+
+1. Next, **extract** the downloaded artifact to a location where we'll be working from.
+
+1. Before the upgrade takes place, we need to apply the data upgrade CRD. You can do this by running `kubectl apply -f /path/to/extract/artifact/dataupgrade-crd.yaml -n cluedin`. You do not need to modify this file.
+
+1. Next, you'll need to update the `platform-custom-values.yaml` patch file to include the neo4j pasword and potentially change the global image tag as it's shipped with `2024.01.00`. This file is part of the extracted artifact. The field is `infrastructure.neo4j.neo4j.password`. It should be set to `xxx` in the file, and failing to update this will potentially break the upgrade process.
```

Comment:
```suggestion
1. Update the `platform-custom-values.yaml` patch file to include the neo4j password and potentially change the global image tag as it's shipped with `2024.01.00`. This file is part of the extracted artifact. The field is `infrastructure.neo4j.neo4j.password`. It should be set to `xxx` in the file, and failing to update this will potentially break the upgrade process.
```